src/lib.rs

```diff
@@ -192,6 +201,13 @@ impl<T> Die<T> for Option<T> {
             None => PrintExit::print_exit(&(exit_code, msg)),
         }
     }
+
+    fn die_with<X: PrintExit>(self, func: fn(()) -> X) -> T {
+        match self {
+            Some(t) => t,
+            None => PrintExit::print_exit(&func(()))
+        }
+    }
```

Comment:
Instead of adding the `die_with` function to the `Die` trait, I would recommend just adding a new trait:
```rust
pub trait DieWith<T, E> {
    fn die_with<X: PrintExit>(self, func: fn(E) -> X) -> T;
}
```
and implementing it only for the `Result<T, E>` type. That way, you don't get the unintuitive syntax for `Option` types:
```rust
None.die_with(|()| ("error message", 2));
```